### PR TITLE
LAZ-15: Laz Cardinal Movement and Sprint

### PR DESCRIPTION
**Pull request by wbeltran777**
- Added simple moving script. Has acceleration, deceleration, and curvature params. Subject to change in the future as we polish it further.
- Press LMB to speed up Laz (this will later also trigger the Lazo)
- Added third party Earth asset as a temporary space like background. 
- Added cinemachine and new input manager. 
- Added Laz prefab and temp materials used for the dummy model

### CODE REVIEW

**Comment by m4rtint** [comment]
Did you pull in third party art from the asset store?

**Comment by wbeltran777** [comment]
Yep from the asset store

File: Assets/Scripts/LazMovementController.cs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    private float _deceleration = 1f;
+    [SerializeField]
+    private float _baseMaxSpeed = 5f;
+    [SerializeField]
+    private float _LazoMaxSpeed = 10f;
+    [SerializeField]
+    [Range(0,1)]
+    private float _curvatureRate = 0.1f;
+
+    private bool _movementPressed = false;
```
Comment (m4rtint) [inline Assets/Scripts/LazMovementController.cs:30]: rename to "_isMovementPressed"

File: Assets/Scripts/LazMovementController.cs [hunk 2]
```diff
@@ -0,0 +1,113 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.InputSystem;
+
+public class LazMovementController : MonoBehaviour
+{
+    private event Action OnLazoActivate;
```
Comment (m4rtint) [inline Assets/Scripts/LazMovementController.cs:10]: We shouldn't be activating Lazo here. 

This class should only be for moving

File: Assets/Scripts/LazMovementController.cs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        else
+        {
+            _inputDirection = Vector2.zero;
+            _movementPressed = false;
+        }
+    }
+
+    public void LazoTrigger(InputAction.CallbackContext context)
+    {
+        // If Lazo is Activated
```
Comment (m4rtint) [inline Assets/Scripts/LazMovementController.cs:102]: This should be another component/script

File: Assets/Scripts/LazMovementController.cs [hunk 4]
```diff
@@ -0,0 +1,113 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.InputSystem;
+
+public class LazMovementController : MonoBehaviour
```
Comment (m4rtint) [inline Assets/Scripts/LazMovementController.cs:8]: Put this in the proper name space

File: Assets/Scripts/LazMovementController.cs [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    private void UpdateRigidBody()
+    {
+        _rigidbody.velocity = _speed * _currentDirection;
+    }
+
+    /// <summary>
+    /// Fired from Player input
+    /// </summary>
+    /// <param name="context"> Data with input information </param>
+    public void UpdateMovementDirection(InputAction.CallbackContext context)
```
Comment (m4rtint) [inline Assets/Scripts/LazMovementController.cs:85]: I'm sure you have the Input Component in the prefavb.  and this method is attched to the component. 

Can you write a note here saying, this method is used in the inspector?

**Comment by m4rtint** [comment]
> Yep from the asset store

I learnt this the hard way but. .

We should not put third party assets in a separate folder. we should leave them imported where ever they are. 
That way it's easier to update/change whenever that's needed.

File: Assets/Scripts/LazMovementController.cs (hunk 2 above)
Comment (wbeltran777) [reply Assets/Scripts/LazMovementController.cs]: Updated

**Comment by wbeltran777** [comment]
Oh shoot that makes sense. Should I reimport it and leave it wherever it imports to?

**Comment by m4rtint** [comment]
> Oh shoot that makes sense. Should I reimport it and leave it wherever it imports to?

yeah sounds good

File: Assets/Scripts/LazMovementController.cs [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.InputSystem;
+
+namespace Laz
+{
+    public class LazMovementController : MonoBehaviour
+    {
+        private event Action OnLazoActivate;
```
Comment (m4rtint) [inline Assets/Scripts/LazMovementController.cs:12]: You can remove this.

**Review by m4rtint — APPROVED** [review]
Lgtm, just make sure this is up to date with master.

**Review by m4rtint — CHANGES_REQUESTED** [review]
Make sure it's up to date with main

**Pull request merged**